mlrun/db/httpdb.py

```diff
@@ -3333,6 +3386,31 @@ def store_datastore_profile(
 
         self.api_call(method="PUT", path=path, body=json.dumps(profile.dict()))
 
+    @staticmethod
+    def _set_artifact_metadata(artifact, key, iter, tag, tree):
+        artifact_dict = artifact if isinstance(artifact, dict) else artifact.to_dict()
```

Comment:
why do you make a dict from the artifact just to turn it back to artifact later? is it for legacy? think we can avoid it or add a comment why

Reply:
To be consistent whether we got an artifact object or a dict

Reply:
removed this function altogether...